Core/CMakeLists.txt

```diff
@@ -24,7 +24,7 @@ target_include_directories(
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 target_link_libraries(
   ActsCore
-  PUBLIC Boost::boost Eigen3::Eigen)
+  PUBLIC Boost::boost Eigen3::Eigen Annoy::Annoy)
```

Comment:
@paulgessinger @asalzburger what do you think about putting this in Core?

Reply:
I also would be worried if we require Annoy as a core dependency. If I understand our policy correctly, The code should go into a plugin, which can have this dependency.

Reply:
Yeah that's a blocker for sure.

Reply:
> I also would be worried if we require Annoy as a core dependency. If I understand our policy correctly, the code should go into a plugin, which can have this dependency.

I understand the concern about having Annoy as a core dependency.

Moving the code to core comes from a discussion with @noemina, allowing it to then be used with Athena.
Is there a way to access the plugins from Athena?

Reply:
Sure we just need to switch the plugin on in the Athena Externals for ACTS